lib/ansible/plugins/inventory/netbox.py

```diff
@@ -246,6 +247,14 @@ def extract_device_role(self, host):
         except Exception:
             return
 
+    def extract_config_contexts(self, host):
+        try:
+            url = urljoin(self.api_endpoint, "/api/dcim/devices/" + str(host["id"]))
+            device_lookup = self._fetch_information(url)
```

Comment:
Why are you doing another fetch? When the host is fetched, the config_context attribute already exists in the host

Reply:
Because host shows only local context not the rendered context. For example-
```
random1": {
                "config_context": [
                    null
                ],
                "device_roles": [
                    "ACI IPN"
                ],
                "device_types": [
                    "random1"
                ],
                "manufacturers": [
                    "Cisco"
                ],
                "racks": [
                    "XYZ"
                ],
                "sites": [
                    "ABC"
                ],
                "tenants": [
                    "ABC"
                ]
            },
            "random": {
                "config_context": [
                    {
                        "cc": {
                            "net": {
                                "vpn_asn": 66000
                            }
                        }
                    }
                ],
                "device_roles": [
                    "ACI IPN"
                ],
                "device_types": [
                    "random"
                ],
                "manufacturers": [
                    "Cisco"
                ],
                "racks": [
                    "XYZ"
                ],
                "sites": [
                    "ABC"
                ],
                "tenants": [
                    "ABC"
                ]
```
However, if we would like to see the rendered context from the source contexts I had to do another fetch.
```
"random1": {
                "config_context": [
                    {
                        "cc": {
                            "net": {
                                "vpn_asn": 65000
                            }
                        }
                    }
                ],
                "device_roles": [
                    "ACI IPN"
                ],
                "device_types": [
                    "random1"
                ],
                "manufacturers": [
                    "Cisco"
                ],
                "racks": [
                    "XYZ"
                ],
                "sites": [
                    "ABC"
                ],
                "tenants": [
                    "ABC"
                ]
            },
            "random": {
                "config_context": [
                    {
                        "cc": {
                            "net": {
                                "vpn_asn": 66000
                            }
                        }
                    }
                ],
                "device_roles": [
                    "ACI IPN"
                ],
                "device_types": [
                    "random"
                ],
                "manufacturers": [
                    "Cisco"
                ],
                "racks": [
                    "XYZ"
                ],
                "sites": [
                    "ABC"
                ],
                "tenants": [
                    "ABC"
                ]
```